drivers/crypto/crypto_tc_shim.c

```diff
@@ -114,24 +114,13 @@ static int do_ccm_encrypt_mac(struct cipher_ctx *ctx,
 	}
 
 	if (tc_ccm_generation_encryption(op->out_buf, op->out_buf_max,
-					 aead_op->ad, aead_op->ad_len, op->in_buf,
-					 op->in_len, &ccm) == TC_CRYPTO_FAIL) {
+				aead_op->ad, aead_op->ad_len, op->in_buf,
+				op->in_len, &ccm, aead_op->tag) == TC_CRYPTO_FAIL) {
 		LOG_ERR("TC internal error during CCM Encryption OP");
 		return -EIO;
 	}
 
-	/* Looks like TinyCrypt appends the MAC to the end of out_buf as it
-	 * does not give a separate hash parameter. The user needs to be aware
-	 * of this and provide sufficient buffer space in output buffer to hold
-	 * both encrypted output and hash
-	 */
-	aead_op->tag = op->out_buf + op->in_len;
-
-	/* Before returning TC_CRYPTO_SUCCESS, tc_ccm_generation_encryption()
-	 * will advance the output buffer pointer by op->in_len bytes,
-	 * and then increment it ccm.mlen times (while writing to it).
-	 */
-	op->out_len = op->in_len + ccm.mlen;
+	op->out_len = op->in_len;
```

Comment:
doing it will cause we have another different behaviour between tc and mbedtls shims. crypto_mtls_shim.c code:
```
	/* This is equivalent to what the TinyCrypt shim does in
	 * do_ccm_encrypt_mac().
	 */
	apkt->pkt->out_len = apkt->pkt->in_len;
	apkt->pkt->out_len += ctx->mode_params.ccm_info.tag_len;
```

as you can see, mtls shim is counting tag_len in output

Reply:
ok, thanks